load-tests/book_reviews_mutation/src/modules/datasource/types.bal

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public type BookRow record {
+    readonly string id;
+    string title;
+    string author;
+};
+
+# Represents a review by a user
+# + bookId - book id
+# + author - author of the review
+# + comment - review comment
+public type ReviewData record {|
+    string bookId;
+    string author;
+    string comment;
+|};
+
+public type ReviewRow record {
+    readonly string id;
+    *ReviewData;
+};
```

Comment:
Can't we make these `readonly & record` types? Then we don't need the `cloneReadOnly` methods.